### DISCUSSION TITLE
Double array wrapping in **call** method

### DISCUSSION THREAD

**Discussion by program-mia** [CodeDredd/laravel-soap#233] [Q&A] [answered]
Hi!

So I just downloaded the package and decided to try out some simple functionality. I have some wsdl that I call a "login" method on, so it has to accept params. I pass parameters as array, so the type check is passed, but then, in SoapClient, on line 357, the arguments passed are wrapped in an array again, which results in wrong parameters passed to the `$this->engine->request` and it throws an error:

> CodeDredd\Soap\Client\Events\ConnectionFailed::__construct(): Argument #1 ($request) must be of type CodeDredd\Soap\Client\Request, null given

I would really love to hear that I am just doing something wrong here. 

Removing line 357
> $arguments = [$arguments];

from the SoapClient solves it and I can actually use the endpoint correctly but I do not understand what I am doing wrong? 

My code looks like this:

```
Soap::withHeaders([
            'content-type' => 'text/html',
            'charset' => 'utf-8'
        ])
            ->baseWsdl(self::ENDPOINT)
            ->login(['login', 'password'])
            ->body();
```

**Comment by CodeDredd** [comment]
😨  .... thx for reporting this. Does this error happen with v3 ?

**Comment by program-mia** [comment, marked as answer]
Yes, it was a fresh V3 install on Laravel 9 project

**Reply by CodeDredd** [reply]
Have you tried?

````php
Soap::withHeaders([
            'content-type' => 'text/html',
            'charset' => 'utf-8'
        ])
            ->baseWsdl(self::ENDPOINT)
            ->login('login', 'password')
            ->body();
````

**Reply by program-mia** [reply]
Sure, if I do so, I get this: 
`CodeDredd\Soap\SoapClient::call(): Argument #2 ($arguments) must be of type Illuminate\Contracts\Validation\Validator|array, string given, called in /Users/mia/Code/Sites/discord-help/vendor/codedredd/laravel-soap/src/SoapClient.php on line 341`

**Reply by program-mia** [reply]
As I understand, now it just breaks at 
`public function call(string $method, Validator|array $arguments = []): Response` because it is still expecting one of the two. I have also tried the Validator approach, but it is still returning an array that is later on wrapped in another array again.

Just to confirm here, when I use the initial approach, that is how it goes:
```
            $arguments = [$arguments];
            $result = $this->engine->request($method, $arguments);
```
The second line here is where the error is produces. If I remove the first, all is working correctly because it's just a one level array and not an array nested in an array

**Reply by CodeDredd** [reply]
ok i get it....so the easy and smart thing is just to add
````php
$arguments = Arr::wrap($arguments);
````

making the change for you....tell me if this change works. thanks.

**Reply by program-mia** [reply]
Just tested that one by changing the code in vendor and yes, this approach will work in that case! 🎉

**Reply by CodeDredd** [reply]
Thanks for testing! 🥇 
-> https://github.com/CodeDredd/laravel-soap/releases/tag/v3.0.1

If you like my work and so on i am always happy about geting a free coffee 😄

**Reply by CodeDredd** [reply]
@program-mia  Seems the fix broke other soap calls....i am adding a new option in the soap config file, so that your soap calls also work. You just need to add this in your config

````php
    'call' => [
        'wrap_arguments_in_array' => false,
    ],
````

**Reply by gelinger777** [reply]
I think we have similar problem . We are on v3.0.2 

```
 'wrap_arguments_in_array' => false,
```

did not helped us .
